libp2p/autotls/acme/utils.nim

```diff
@@ -0,0 +1,24 @@
+import base64, strutils, chronos/apps/http/httpclient, json
+import ../../errors
+
+type ACMEError* = object of LPError
+
+proc base64UrlEncode*(data: seq[byte]): string =
+  ## Encodes data using base64url (RFC 4648 §5) — no padding, URL-safe
+  var encoded = base64.encode(data, safe = true)
+  encoded.removeSuffix("=")
+  encoded.removeSuffix("=")
+  return encoded
+
+proc getResponseBody*(
+    response: HttpClientResponseRef
+): Future[JsonNode] {.async: (raises: [ACMEError]).} =
+  try:
+    let responseBody = bytesToString(await response.getBodyBytes()).parseJson()
+    return responseBody
```

Comment:
because  of `await`, code must raise `CatchableError` without converting it to `ACMEError`

Reply:
yes :)